File: fib-seq.js
var recursive_fib = function (n) {
  if (n===1) 
  {
    return [0, 1];
  } 
  else 
  {
    var s = recursive_fib(n - 1);
    s.push(s[s.length - 1] + s[s.length - 2]);
    return s;
  }
};

 console.log(recursive_fib(35));

 var iterative_fib = function(n) {
  let arr = [0, 1];
  for (let i = 2; i < n + 1; i++){
    arr.push(arr[i - 2] + arr[i -1])
  }
 return arr[n]
};

console.log(iterative_fib(35));

File: fib-seq.rb
require 'benchmark'

def recursive_fib(n)
  if (n < 2)
    return n
  else
    return recursive_fib(n - 1) + recursive_fib(n - 2)
  end
end

puts recursive_fib(35)

def iterative_fib(n)
  arr = [0, 1]
  for i in 2..n do
    arr.push(arr[i - 2] + arr[i -1])
  end
 return arr[n]
end

puts iterative_fib(35)


num = 35
Benchmark.bm do |x|
  x.report("recursive_fib") { recursive_fib(num) }
  x.report("iterative_fib")  { iterative_fib(num)  }
end


